botorch/acquisition/analytic.py

```diff
@@ -1068,24 +1067,40 @@ def _get_noiseless_fantasy_model(
             "Only SingleTaskGP models with known observation noise "
             "are currently supported for fantasy-based NEI & LogNEI."
         )
+
     # initialize a copy of SingleTaskGP on the original training inputs
     # this makes SingleTaskGP a non-batch GP, so that the same hyperparameters
     # are used across all batches (by default, a GP with batched training data
     # uses independent hyperparameters for each batch).
+    outcome_transform = deepcopy(getattr(model, "outcome_transform", None))
     fantasy_model = SingleTaskGP(
         train_X=model.train_inputs[0],
         train_Y=model.train_targets.unsqueeze(-1),
         train_Yvar=model.likelihood.noise_covar.noise.unsqueeze(-1),
+        covar_module=deepcopy(model.covar_module),
+        mean_module=deepcopy(model.mean_module),
+        outcome_transform=outcome_transform,
+        input_transform=deepcopy(getattr(model, "input_transform", None)),
```

Comment:
`model.train_inputs[0]` and `model.train_targets` are already in the transformed space, so I don't think we want to  (re)apply the transforms here.

Reply:
It is true that in the SingleTaskGP constructor, it will transform the already transformed train_targets (but not the train_inputs[0]). But that doesn't matter, because the data is reset afterward.
But okay, sure, I guess it's unnecessary to apply them yet again.

Also by the way, I found the problem - it was a subtle problem with the testing code. I fixed it and now all the tests pass! However I needed to increase one of the error tolerances for comparing the gradients in float32 case when RBFKernel was used instead of MaternKernel - probably because, for some reason, RBFKernel gave numerical ill-conditioning warnings but MaternKernel didn't, I'm not sure why.